src/usecases/uc_get_por_email.py

```diff
@@ -0,0 +1,17 @@
+from src.interfaces.IRepoUsuario import IArmazenamento
+from src.usecases.erros.erros_uc_alteracao_info_cadastro import ErroUsuarioInvalido
+
+
+class UCGetPorEmail:
+    usuarioRepo: IArmazenamento
+
+    def __init__(self, usuarioRepo: IArmazenamento):
+        self.usuarioRepo = usuarioRepo
+
+    def __call__(self, email: str):
+        # Email nao pode ser validado pelo model de email - nao será feito erro de badrequest
+        # Erro é re-levantado pois repo pode ser alterado
```

Comment:
Se for re-levantar o erro, levanta o mesmo do repo. Nesse caso, `ErroUsuarioNaoEncontrado`, usando `except Exception as err: raise err`.

Porém, pra levantar o mesmo erro, é só chamar a funcao sem `try/except`. O erro sera carregado por toda a callstack, desde o repo ate a main.

Apesar que nao entendi pq o repo deixaria de levanta-lo se fosse alterado. Quando o repo volatil for substituido pelo SQL, ele obrigatoriamente tera que levantar os mesmos erros de antes, com acrescimo de erros especificos da tecnologia (erro de conexao, erro de query, erro de leitura etc).